src/lookup.py

```diff
@@ -540,15 +588,62 @@ def getMirrorEntity(s):
                 ):
                     self.kinds[entitykind].append(entitygroupname)
 
-            params = entity.findall("param")
-            self.hasParameters = len(params) != 0
-            self.parameters = list(map(EntityLookup.EntityConfig.Parameter, params))
+            bitfields = entity.findall("bitfield")
+            self.hasBitfields = len(bitfields) != 0
+            self.bitfields = list(map(EntityLookup.EntityConfig.Bitfield, bitfields))
 
-        def clampParameterValues(self, subtype):
-            for param in self.parameters:
-                subtype = param.clampValue(subtype)
+        def hasBitfieldKey(self, key):
+            for bitfield in self.bitfields:
+                if bitfield.key == key:
+                    return True
 
-            return subtype
+            return False
+
+        def getBitfieldElements(self):
+            elements = []
+            for bitfield in self.bitfields:
+                elements.extend(bitfield.elements)
+
+            return elements
+
+        def isOutOfRange(self):
+            if self.type >= 1000 and not self.isGridEnt:
+                return True
+            if self.variant >= 4096 and not self.hasBitfieldKey("Variant"):
+                return True
+            if self.subtype >= 255 and not self.hasBitfieldKey("Subtype"):
+                return True
+
+            return False
+
+        def getEditorWarnings(self):
+            warnings = ""
+            if self.type >= 1000 and not self.isGridEnt:
+                warnings += "\nType is outside the valid range of 0 - 999! This will not load properly in-game!"
+            if self.variant >= 4096 and not self.hasBitfieldKey("Variant"):
+                warnings += "\nVariant is outside the valid range of 0 - 4095!"
+            if self.subtype >= 255 and not self.hasBitfieldKey("Subtype"):
+                warnings += "\nSubtype is outside the valid range of 0 - 255!"
+            if self.invalid:
+                warnings += "\nMissing entities2.xml entry! Trying to spawn this WILL CRASH THE GAME!!"
+            if self.hasBitfields:
+                warnings += "\nMiddle-click to configure entity properties"
+
+            return warnings
```

Comment:
would it make sense to consolidate these functions in some way? Maybe by having isOutOfRange return an enum?